### PR TITLE
PMM-7194 force refresh share panels

### PR DESCRIPTION
**Pull request by lunaticusgreen**
**What this PR does / why we need it**: Disable lazy loading for panel share buttons

**Which issue(s) this PR fixes**:  PMM-7194

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 1]
```diff
@@ -260,16 +260,20 @@ export class DashboardModel {
     dispatch(onTimeRangeUpdated(timeRange));
   }
 
-  startRefresh() {
+  startRefresh(forceRefresh?: boolean) {
```
Comment (atymchuk) [inline public/app/features/dashboard/state/DashboardModel.ts:263]: We could avoid this parameter, as we have seen in https://github.com/percona-platform/grafana/pull/137/files

File: public/app/features/dashboard/state/PanelModel.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ import { CoreEvents } from '../../../types';
 
 export const panelAdded = eventFactory<PanelModel | undefined>('panel-added');
 export const panelRemoved = eventFactory<PanelModel | undefined>('panel-removed');
+export const snapshotCreated = eventFactory('snapshot-created');
```
Comment (atymchuk) [inline public/app/features/dashboard/state/PanelModel.ts:31]: This needs to follow the pattern above
```suggestion
export const snapshotCreated = eventFactory<PanelModel | undefined>('snapshot-created');
```
```suggestion
export const snapshotCreated = eventFactory('snapshot-created');
```

File: public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx [hunk 3]
```diff
@@ -80,11 +80,15 @@ export class ShareSnapshot extends PureComponent<Props, State> {
...
+    // @ts-ignore
+    window.forceRefresh = true;
     this.setState({ isLoading: true });
-    this.dashboard.startRefresh();
+    this.dashboard.startRefresh(true);
 
     setTimeout(() => {
       this.saveSnapshot(this.dashboard, external);
+      // @ts-ignore
+      window.forceRefresh = false;
```
Comment (atymchuk) [inline public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx:91]: We can do as well w/o this attribute, as we saw in https://github.com/percona-platform/grafana/pull/137/files

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: public/app/features/dashboard/state/DashboardModel.ts (hunk 1 above)
Comment (tiagomotasantos) [reply public/app/features/dashboard/state/DashboardModel.ts]: Done 👍

File: public/app/features/dashboard/state/PanelModel.ts (hunk 2 above)
Comment (tiagomotasantos) [reply public/app/features/dashboard/state/PanelModel.ts]: Done 👍

File: public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx (hunk 3 above)
Comment (tiagomotasantos) [reply public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx]: Done 👍

**Comment by lunaticusgreen** [comment]
@tiagomotasantos looks great, please move it further

**Review by tiagomotasantos — APPROVED** [review]

**Review by tiagomotasantos — APPROVED** [review]

**Review by atymchuk — APPROVED** [review]

**Pull request merged**